tsdb/chunks/head_chunks.go

```diff
@@ -577,7 +577,10 @@ func (cdm *ChunkDiskMapper) IterateAllChunks(f func(seriesRef, chunkRef uint64,
 
 			idx += ChunkEncodingSize // Skip encoding.
 			dataLen, n := binary.Uvarint(mmapFile.byteSlice.Range(idx, idx+MaxChunkLengthFieldSize))
-			idx += n + int(dataLen) // Skip the data.
+			idx += n
+
+			numSamples := binary.BigEndian.Uint16(mmapFile.byteSlice.Range(idx, idx+2))
```

Comment:
That is how our XOR chunks is designed as I see it https://github.com/prometheus/prometheus/blob/fd4521d2ebac6e467dd35db1e38c0e97aad49921/tsdb/chunkenc/xor.go#L77-L80

Reply:
yeah, now I see it. we should really extend the doc to explain the format of the data field inside the chunks as it would be useful. Anyway it is not related to this PR, but thanks for the pointer.